neothesia/src/scene/playing_scene/midi_player.rs

```diff
@@ -198,69 +252,201 @@ pub enum MidiEventSource {
 struct UserPress {
     timestamp: Instant,
     note_id: u8,
+    time_key_up: Instant,
```

Comment:
Sounds like this should be `Option<Instant>`, and every case of `timestamp == time_key_up` should be just replaced with `time_key_up.is_none()`

It should start as `None` and when the note ends it should become `Some(time)`